### PR TITLE
fix: include OTel semconv 1.20 attributes in spans

### PR DESCRIPTION
**Pull request by robbkidd**
## Which problem is this PR solving?

The 1.21+ HTTP attributes are not yet stable. Per [the HTTP semantic convention v1.21/v1.22 guidance](https://opentelemetry.io/docs/specs/semconv/http/http-spans/), we should send the "old" style attributes. Doing so will help the agent's telemetry play better with others when appearing alongside application instrumentation in the same set of services.

## Short description of the changes

- Update handler to also include the v1.20 semconv attributes other OTel instrumentation is currently using so that existing queries match.

### Proposed future followup:
- Add the `OTEL_SEMCONV_STABILITY_OPT_IN` flag for opting in to the new, as-yet unstable attribute names?

### CODE REVIEW

**Comment by robbkidd** [comment]
We reckon we can decide on whether/how to implement the `OTEL_SEMCONV_STABILITY_OPT_IN` flag in a followup!

**Comment by robbkidd** [comment]
Might have gotten carried away with refactoring the code to make unit testing simpler. 😬

File: handlers/otel_handler.go [hunk 1]
```diff
@@ -103,90 +102,102 @@ func (handler *otelHandler) handleEvent(event assemblers.Event) {
...
 }
 
-func (handler *otelHandler) createHTTPSpan(event *assemblers.HttpEvent, startTime, endTime time.Time, attrs []attribute.KeyValue) {
+func (handler *otelHandler) createHTTPSpan(event *assemblers.HttpEvent, startTime, endTime time.Time, incomingAttrs []attribute.KeyValue) {
 	var spanName string
 	if event.Request() == nil {
 		spanName = "HTTP"
 	} else {
-		spanName = fmt.Sprintf("HTTP %s", event.Request().Method)
+		spanName = event.Request().Method
```
Comment (robbkidd) [inline handlers/otel_handler.go:110]: [HTTP span name should be one of the recognized methods](https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name). The literal string "HTTP" is only used as a fallback name with the method in the request is something non-standard.

**Review by MikeGoldsmith — COMMENTED** [review]
I like what your refactoring is doing. Maybe it could go one step further so the `handleEvent` func is responsible for creating the span, then we have `handleHTTPEvent` funcs that gather the specific event type attrs and the span name. This will make adding more event type handlers easier / cleaner.

Something like:
```
func handleHTTPEvent(event) (spanName string, attrs []attribute.keyvalye)
```

**Comment by robbkidd** [comment]
> Maybe it could go one step further … 

😈 You read my mind, @MikeGoldsmith, but I held off because I thought I was already going pretty far in this refactor.

I considered again having a `handleHTTPEvent()` return a span name and slice of attributes, but I think Not Right Now™. There's some `http.` attrs we set in the ~wibblyWobblyTimeyWimey~ `getEventStartEndTimestamps()` function. Maybe further handler logic separate can happen in a followup PR, maybe when we've got a second network protocol to handle.

File: handlers/otel_handler_test.go [hunk 2]
```diff
@@ -55,3 +56,54 @@ func TestHeaderToAttributes(t *testing.T) {
...
+	t.Run("a realish request", func(t *testing.T) {
+		attrs := defaultHandler.resolveHTTPAttributes(realishEvent)
+
+		assert.Contains(t, attrs, attribute.String("http.request.method", "GET"))
+		assert.Contains(t, attrs, attribute.String("http.method", "GET"))
+		assert.Contains(t, attrs, attribute.String("http.request.header.user_agent", "teapot-checker/1.0"))
+		assert.Contains(t, attrs, attribute.String("http.request.header.connection", "keep-alive"))
+		assert.Contains(t, attrs, attribute.Int("http.response.status_code", 418))
+		assert.Contains(t, attrs, attribute.Int("http.status_code", 418))
+		assert.Contains(t, attrs, attribute.String("error", "HTTP client error"))
```
Comment (JamieDanielson) [inline handlers/otel_handler_test.go:89]:
```suggestion
		assert.Contains(t, attrs, attribute.String("error", "HTTP client error"))
		assert.Contains(t, attrs, attribute.Int("http.request_content_length", 42))
		assert.Contains(t, attrs, attribute.Int("http.request.body.size", 42))
		assert.Contains(t, attrs, attribute.Int("http.response_content_length", 84))
		assert.Contains(t, attrs, attribute.Int("http.response.body.size", 84))
```

A few more dual-send http attributes that could be included in here

**Review by JamieDanielson — APPROVED** [review]
This is great! I suggested a few more attributes to a test case but then I think this is good to go!

**Comment by JamieDanielson** [comment]
Also noting this table for quick reference, in preparation for docs updates:

"http.request.method" / "http.method"
"http.response.status_code" / "http.status_code"
"http.request.body.size" / "http.request_content_length"
"http.response.body.size" / "http.response_content_length"
"url.path" / "http.target"

**Pull request merged**